lib/ansible/plugins/test/core.py

```diff
@@ -109,8 +109,7 @@ def started(result):
         raise errors.AnsibleFilterError("The 'started' test expects a dictionary")
     if 'started' in result:
         # For async tasks, return status
-        # NOTE: The value of started is 0 or 1, not False or True :-/
-        return result.get('started', 0) == 1
+        return result.get('started', False)
```

Comment:
These results should be explicitly `bool`ified in place- DT TemplateEngine will fire a deprecation warning if a test returns a non-boolean result (eg, if you got the raw result from an older version of the Windows module without going through the action or something else).

Reply:
yes, counting on that to trip the tests instead of relying on `True == 1` to handle both cases, but I'll have to revert anyways if this covers windows testing also since we cannot ensure the return is boolean now.

Reply:
In 2.19+, if a Jinja test impl returns anything other than a bool, it fires a deprecation warning. These need to always return bools- if you want something custom for testing, it needs to be implemented somewhere else.

Reply:
see the `type_debug == 'bool'` for explicit boolean testing, I kept/added ` x == 1` and `x == True` to ensure the variations work

Reply:
rebased to include your 'bool()' additions to all tests